### PR TITLE
Bug/static command translation

### PR DESCRIPTION
**Pull request by Mylan719**
#906

### CODE REVIEW

**Review by quigamdev — APPROVED** [review]
I have tested these changes on a commercial project where this issue appeared and it worked as expected. Also thanks for the refactoring and comments. It is easier to understand what is happening there.

**Review by exyi — APPROVED** [review]
Very nice :)

File: src/DotVVM.Framework.Tests.Common/Binding/StaticCommandCompilationTests.cs [hunk 1]
```diff
@@ -57,80 +79,223 @@ public string CompileBinding(string expression, Type[] contexts, Type expectedTy
...
+            Assert.AreEqual("(function(a,c,b){return new Promise(function(resolve,reject){dotvvm.staticCommandPostback(a,\"WARNING/NOT/ENCRYPTED+++WyJEb3RWVk0uRnJhbWV3b3JrLlRlc3RzLkJpbmRpbmcuVGVzdFNlcnZpY2UsIERvdFZWTS5GcmFtZXdvcmsuVGVzdHMuQ29tbW9uIiwiTG9hZCIsW10sIkFRQUEiXQ==\",[c.$data.StringProp(),c.$data.StringProp()],options).then(function(r_0){(b=(c.$data.StringProp(r_0).StringProp(),\"Test\"),dotvvm.staticCommandPostback(a,\"WARNING/NOT/ENCRYPTED+++WyJEb3RWVk0uRnJhbWV3b3JrLlRlc3RzLkJpbmRpbmcuVGVzdFNlcnZpY2UsIERvdFZWTS5GcmFtZXdvcmsuVGVzdHMuQ29tbW9uIiwiTG9hZCIsW10sIkFRQT0iXQ==\",[c.$data.StringProp()],options).then(function(r_1){resolve((b,c.$data.StringProp(r_1).StringProp()));},reject));},reject);});}(this,ko.contextFor(this)))", result);
+        }
+
+        [TestMethod]
+        public void StaticCommandCompilation_ExpressionBetweenPostbacks_NoParametersLast()
+        {
+            var result = CompileBinding("StringProp = injectedService.Load(IntProp); \"Test\"; StringProp2 = injectedService.Load()", niceMode: true, new[] { typeof(TestViewModel) });
```
[the remark this answers is not in the document]
Comment (Mylan719) [reply src/DotVVM.Framework.Tests.Common/Binding/StaticCommandCompilationTests.cs:167]: I tried to remedy the issue by taking all the whitespaces out of both the result and the expected code. Then I compare them.

[the remark this answers is not in the document]
Comment (exyi) [reply src/DotVVM.Framework.Tests.Common/Binding/StaticCommandCompilationTests.cs:167]: Aha, I have missed that, that's nice.

**Pull request merged**